kartothek/core/common_metadata.py

```diff
@@ -78,6 +77,35 @@ def _schema_compat(self):
 
             schema = schema.remove_metadata()
             md = {b"pandas": _dict_to_binary(pandas_metadata)}
+            # https://github.com/JDASoftwareGroup/kartothek/issues/259
+
+            if schema is not None:
+                fields = []
+                for f in schema:
```

Comment:
short comment before the entire massaging would be nice. This function just does more and more and it is hard to follow along.